pmd-java/src/main/resources/category/java/codestyle.xml

```diff
@@ -1952,6 +1952,7 @@ which makes the code also more readable.
                 <value>
                     <![CDATA[
 //VariableInitializer
+/../..[(Type)]
```

Comment:
I wonder if this change is causing multiple reports on lines that actually should have only 1 violation:

For instance
```java
List<String> foo, bar = new ArrayList<String>(), baz = new ArrayList<>();
```

should produce 1 violation at that line, but I *think* this change makes it report 2 times

Reply:
Rule Designer is matching the node only once for **bar = new ArrayList<String>()** but test is reporting twice. Why is **@AnonymousClass='true'** showing error in matched nodes but only **@AnonymousClass** is working fine in designer.

Reply:
@Shubham-2k17 Probably because the XPath version you used with the designer was 2.0. In XPath 2.0, `@AnonymousClass = "true"` is a type error, rather one should use `@AnonymousClass = true()`. The rule is XPath 1.0 which explains why that works. This weird behaviour is a consequence of #1244. You can set the XPath version of the designer with the cog icon above the XPath area (since 6.11.0, otherwise it's a choicebox).

@jsotuyod is right, the node is matched twice:
![capture du 2019-02-07 13-22-24](https://user-images.githubusercontent.com/24524930/52411308-a7a00f80-2adb-11e9-8282-d0598467e029.png)

I suggest you use the `@TypeInferred` attribute of the variable declarator id, eg
```ruby
//VariableInitializer[preceding-sibling::VariableDeclaratorId[1]/@TypeInferred="false"]
```
instead of checking for the presence of a type node.

Reply:
@Shubham-2k17 awesome to see you updated this! We are probably missing a test case non the less, as the build passed with the alternative that reported twice.

After that, I think this is good to merge